### DISCUSSION TITLE
SensESP and ESP32 Devkit randomly freezing

### DISCUSSION THREAD

**Discussion by johnthebairdsca** [SignalK/SensESP#647] [General]
Hello,

I am a full on noob and need a little bit guidance on how I go about troubleshooting an issue with my ESP32 Devkit running sensESP whereby it randomly freezes. Sometimes within 30 seconds, other times after a few mins…

In vscode observing the terminal window through the “load and monitor” process… all appears fine… I see output in the terminal window… then the messages stop and the blue led goes off and the red led stays on and the ESP32 is non-reaponsive and either needs to be disconnected from the power and plugged back in OR I go through a load and monitor to get it working again. 

Thoughts / suggestions?

Thanks!

John

**Comment by ba58smith** [comment]
@johnthebairdsca, I suggest you take this to the #sensors channel on the Signal K Slack: https://signalk-dev.slack.com/join/shared_invite/zt-1leccop43-KrU7G6yBq9g91KXjZtNg1g#/shared-invite/email

And when you post your question there, include your entire main.cpp file contents.

**Reply by johnthebairdsca** [reply]
Thx!

**Comment by mairas** [comment]
Personally, I actually quite prefer GitHub Discussions because the message threads are archived and can benefit future generations as well. But Butch is right that you might get a bigger audience at Slack. And in any case, you'd need to include a bit of context of what you're trying to do, what you see on the serial monitor and the hardware setup is.

**Reply by johnthebairdsca** [reply]
No problem and thanks guys. 

Will do a little more mucking around on my end and pending how that goes along with my schedule this week will more than likely get something posted this weekend.
